src/user.ts

```diff
@@ -1104,7 +1113,7 @@ export class User {
       req.user.provider,
       user
     );
-    await this.userDB.insert(finalUser);
```

Comment:
This completely unrelated change would introduce a security vulnerability.

Reply:
Okay.  But I didn't see a way to verify that the session belonged to the requestor user_id.  I'll look at your code again so that I can implement my own very secure implementation.  
Profile information is implemented via Oauth, right?  Why would that not be requisite functionality for local accounts?

Reply:
private async completeEmailChange doesn't seem to validate that the session is owned by the requestor.  Theoretically, I could initiate a session and change anyone's email.  How is this handled internally to prevent spoofing?  That's what I'm not seeing.